Feature: Allow Custom Content for Each Instance

ardelato opened the pull request:
## Description

This does a couple things that essentially allow us to _extend_ Laravel's `lang/` logic to allow instance based content.

It moves all current translation files one level deeper into `instance` subgroups. Then by utilizing an environment variable, we can specify which content we'd want to implement.

### CR Notes

To get this to work it required a custom translation service provider and dropping [lang.js](https://github.com/rmariuzzo/Lang.js) for a custom implementation.

qa_req 0

ardelato committed:
feat(lang): create instance localization

ardelato committed:
feat(translations): implement instance-aware translation loader

Added a custom TranslationServiceProvider to support instance-specific translations.
This will extend Laravel's lang structure to have separate translations
for each Restarters instance. As such, this allows us to retain the
translations functionality but allow content to be overridden for each instance.

While doing so I merged the RobustTranslator code into the TranslationServiceProvider,
to remove the file and contain all the logic within the service provider.

This also updates the .env files to include APP_INSTANCE variable for
dynamic instance configuration.

ardelato committed:
feat(vue): add instance-aware script to header templates

With the addition of the APP_INSTANCE environment variable, we will need
to pass the env from the Laravel app to the Vue components. As such,
we will add a script to both header templates to set the window.appInstance
variable to then be used by the Vue components.

ardelato committed:
refactor(vue-lang): create custom lang.js

We have a file named lang.js but it uses an actual package called
lang.js that has not been updated for a couple of years now.
https://github.com/rmariuzzo/Lang.js

On top of that, the inner workings of the package no longer work with our new
lang folder structure that is subgrouped by instances. The generated
translations.js file has a structure of

instances.<instance>.<locale>.<group|key> : string|object

The lang.js package splits the key by the dots and assumes each part
is a new level of a nested object. As such, it expects 'instances' to
be a key pointing to an object while it should really be
instances.instance.locale.group|key. as the first level of the lookup.

To resolve this, we will create a custom lang.js file, that builds on the
original lang.js package, and allows us to lookup translations for
our given structure.

To also avoid confusion, the lang.js file will house the LangJS class
and a new file i18n.js will be created to house the original __() translation
functionality.

ardelato committed:
feat(debug): add appDebug variable for translation debugging

Let's make use of the debug option for lang.js to log missing translations.
We can utilize APP_DEBUG to enable this feature.

That said, there was an issue with the boolean value being returned from
the PHP env() function. It returns 1 for true, 0 for false instead of
true/false.

ardelato committed:
feat(translations): add custom language files for the Fixit Clinic instance

ardelato committed:
feat(errors): add error reporting translations for instances

Created new language files for error reporting in the base and Fixit Clinic instances.
This will allow us to customize the error reporting for each instance.
To actually render the HTML in the error translations we need to use the
{!! !!} syntax to not escape the HTML.

Additionally, it seems there was a bug that was introduced with our
Laravel Shift upgrade that introduced raw text
"use App\Providers\AppServiceProvider;" when in reality the Laravel
Shift upgrade was trying to utilize the AppServiceProvider.php file.
However, these files do not need to use the AppServiceProvider.php file
so we can remove it and therefore remove the raw text from being rendered
on the error pages.

ardelato committed:
refactor(env): replace APP_LOGO with APP_INSTANCE

Instead of having two different env vars that essentially do the same
thing, we can just use one.

djmetzle reviewed (approved):
This is an interesting approach to multi-tenancy. CR :flags:

ardelato committed:
fix(docker): run npm production after translation generation

The public/js/translations.js file was not correct because we were
copying the translations.js file from resources/js/translations.js before
we ran the lang:js artisan command. As such, the translations.js file
was out of sync. We need to run npm production after the lang:js
artisan command to ensure the translations.js file is correct.

ardelato commented:
deploy_block ⚡ on #13

ardelato commented:
un_deploy_block ⚡ #13 is out

ardelato committed:
fix(logo): update logo inclusion logic

If the APP_INSTANCE is set to `base`, then the restarters logo won't be
used since there is no logo-base.blade.php file. I chose not to add or
rename the restarters logo file to logo-base.blade.php since its not
really a base logo. Instead, I chose to update the conditional logic
to use restarters in both scenarios. Meaning, if the APP_INSTANCE is
set to base, then the restarters logo will be used, as well if APP_INSTANCE
is not set at all.

ardelato committed:
feat(landing): update landing page content and layout

Modified the landing page language file to reflect new messaging for the
Repair Event Locator. That said, the new content required an adjustment
to the layout to better fit the new content.

djmetzle reviewed (approved):
CR :flags:
